staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -212,6 +217,27 @@ func newAuthenticator(c *cache, config *api.ExecConfig, cluster *clientauthentic
 	return c.put(key, a), nil
 }
 
+func isInteractive(isTerminalFunc func(int) bool, config *api.ExecConfig) (bool, error) {
+	if config.InteractiveMode == nil {
+		// backwards compatible default
+		return isTerminalFunc(int(os.Stdout.Fd())), nil
```

Comment:
It's debatable if we want preserve this (incorrect?) behavior.

Reply:
Did we decide over Zoom that we would need to preserve this (incorrect) behavior for the sake of v1beta1 backwards compatibility? For v1 I think we decided we want to use `os.Stdin.Fd()` only.

Reply:
Fixed this to match 1.21 bug fix behavior: https://github.com/kubernetes/kubernetes/pull/99310/commits/4acb159638e01eb707bf8247dd6e88bcf2f29969.

Reply:
resolved by IfAvailable approach which matches old behavior and is a generally useful thing for a plugin to be able to express